ffi/capi/src/exemplar_chars.rs

```diff
@@ -47,6 +48,28 @@ pub mod ffi {
             self.0.as_borrowed().contains32(cp)
         }
 
+        /// Get an iterator of all the code point ranges in the current exemplar character set.
+        #[diplomat::rust_link(
+            icu::collections::codepointinvliststringlist::CodePointInversionListAndStringList::code_points,
+            FnInStruct
+        )]
+        pub fn code_point_ranges<'a>(&'a self) -> Box<CodePointRangeIterator<'a>> {
+            let ranges = self.0.as_borrowed().code_points().iter_ranges().collect::<Vec<_>>();
+
+            Box::new(CodePointRangeIterator(Box::new(ranges.into_iter())))
+        }
+
+        /// Get an iterator of all the code point ranges in the current exemplar character set.
+        #[diplomat::rust_link(
+            icu::collections::codepointinvliststringlist::CodePointInversionListAndStringList::strings,
+            FnInStruct
+        )]
+        pub fn strings(&self) -> Box<StringIterator> {
+            let strings = self.0.as_borrowed().strings().iter().map(|s| s.to_string()).collect::<Vec<_>>();
```

Comment:
issue: these two should not collect, Diplomat can return borrows. This should return a `Box<ExemplarCharactersStringIterator<'a>>`

Ideally there's no dynamic dispatch, if we need concrete iterator types in ICU4X we should add them.